@google-labs-jules when I switch to this branch and run the app, I get the default data in the UI and this error in the console when I navigate from the StopsScreen to the StopScreen:
```
console.js:668 Running "CATrackerFresh" with {"rootTag":1,"initialProps":{},"fabric":true}
console.js:668 ProgressBarAndroid has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-community/progress-bar-android' instead of 'react-native'. See https://github.com/react-native-progress-view/progress-bar-android
anonymous @ console.js:668
overrideMethod @ backend.js:17416
anonymous @ setUpDeveloperTools.js:42
registerWarning @ LogBox.js:247
anonymous @ LogBox.js:113
warnOnce @ warnOnce.js:27
get ProgressBarAndroid @ index.js:82
metroImportAll @ require.js:149
importAll @ asyncRequire.js:18
console.js:668 SafeAreaView has been deprecated and will be removed in a future release. Please use 'react-native-safe-area-context' instead. See https://github.com/AppAndFlow/react-native-safe-area-context
anonymous @ console.js:668
overrideMethod @ backend.js:17416
anonymous @ setUpDeveloperTools.js:42
registerWarning @ LogBox.js:247
anonymous @ LogBox.js:113
warnOnce @ warnOnce.js:27
get SafeAreaView @ index.js:101
metroImportAll @ require.js:149
importAll @ asyncRequire.js:18
console.js:668 Clipboard has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-clipboard/clipboard' instead of 'react-native'. See https://github.com/react-native-clipboard/clipboard
anonymous @ console.js:668
overrideMethod @ backend.js:17416
anonymous @ setUpDeveloperTools.js:42
registerWarning @ LogBox.js:247
anonymous @ LogBox.js:113
warnOnce @ warnOnce.js:27
get Clipboard @ index.js:190
metroImportAll @ require.js:149
importAll @ asyncRequire.js:18
console.js:668 InteractionManager has been deprecated and will be removed in a future release. Please refactor long tasks into smaller ones, and  use 'requestIdleCallback' instead.
anonymous @ console.js:668
overrideMethod @ backend.js:17416
anonymous @ setUpDeveloperTools.js:42
registerWarning @ LogBox.js:247
anonymous @ LogBox.js:113
warnOnce @ warnOnce.js:27
get InteractionManager @ index.js:236
metroImportAll @ require.js:149
importAll @ asyncRequire.js:18
console.js:668 PushNotificationIOS has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-community/push-notification-ios' instead of 'react-native'. See https://github.com/react-native-push-notification/ios
anonymous @ console.js:668
overrideMethod @ backend.js:17416
anonymous @ setUpDeveloperTools.js:42
registerWarning @ LogBox.js:247
anonymous @ LogBox.js:113
warnOnce @ warnOnce.js:27
get PushNotificationIOS @ index.js:292
metroImportAll @ require.js:149
importAll @ asyncRequire.js:18
migrations.ts:198 [db] Migrations completed
Bootstrap.tsx:56 Bootstrap complete
Welcome to React Native DevTools
Debugger integration: Android Bridgeless (ReactHostImpl)
console.js:668 Error: Uncaught (in promise, id: 0): "Error: app.db is not open"
    at onUnhandled (10.0.2.2:8081/index.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=false&runModule=true&excludeSource=true&sourcePaths=url-server:23279:59)
    at onUnhandled (address at InternalBytecode.js:1:5443)Caused by: Error: app.db is not open
    at close (native)
    at anonymous (10.0.2.2:8081/node_modules/react-native-quick-sqlite/lib/module/index.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=true&runModule=false&excludeSource=true&sourcePaths=url-server:99150:11)
    at ?anon_0_ (10.0.2.2:8081/src/db/native/QuickSQLiteAdapter.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=true&runModule=false&excludeSource=true&sourcePaths=url-server:196:79)
    at next (native)
    at asyncGeneratorStep (10.0.2.2:8081/index.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=false&runModule=true&excludeSource=true&sourcePaths=url-server:24706:19)
    at _next (10.0.2.2:8081/index.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=false&runModule=true&excludeSource=true&sourcePaths=url-server:24720:29)
    at anonymous (10.0.2.2:8081/index.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=false&runModule=true&excludeSource=true&sourcePaths=url-server:24725:14)
    at tryCallTwo (address at InternalBytecode.js:1:1307)
    at Promise (address at InternalBytecode.js:1:1418)
    at anonymous (10.0.2.2:8081/index.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=false&runModule=true&excludeSource=true&sourcePaths=url-server:24717:25)
    at apply (native)
    at close (10.0.2.2:8081/src/db/native/QuickSQLiteAdapter.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=true&runModule=false&excludeSource=true&sourcePaths=url-server:202:30)
    at anonymous (10.0.2.2:8081/index.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=false&runModule=true&excludeSource=true&sourcePaths=url-server:110311:23)
    at next (native)
    at asyncGeneratorStep (10.0.2.2:8081/index.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=false&runModule=true&excludeSource=true&sourcePaths=url-server:24706:19)
    at _next (10.0.2.2:8081/index.bundle//&platform=android&dev=true&lazy=true&minify=false&app=com.catrackerfresh&modulesOnly=false&runModule=true&excludeSource=true&sourcePaths=url-server:24720:29)
    at tryCallOne (address at InternalBytecode.js:1:1274)
    at anonymous (address at InternalBytecode.js:1:4865)
anonymous @ console.js:668
overrideMethod @ backend.js:17416
reactConsoleErrorHandler @ ExceptionsManager.js:184
anonymous @ setUpDeveloperTools.js:42
reportException @ ExceptionsManager.js:108
handleException @ ExceptionsManager.js:173
onUnhandled @ promiseRejectionTrackingOptions.js:46

```
I do not get a similar error going from the RouteScreen to the StopScreen. 
Do you know what's going wrong? I had previously had `await db.close();`  but I removed the calls because I had identified those calls as what was triggering a similar error.

--- a/src/RouteScreen.tsx
+++ b/src/RouteScreen.tsx
@@ -11,20 +11,16 @@ const DEFAULT_STOPS: StopItem[] = [{stopId: '0', stopName: 'Stop 0'}];
 
 async function loadStops(busRouteId: string) {
     const db = await openDatabase({ name: 'app.db' });
-    try {
-        const stopItems: StopItem[] = [];
-        const stops = await db.execute('SELECT rs.stop_id, stops.stop_name FROM route_stops rs JOIN stops ON rs.stop_id = stops.stop_id WHERE route_id = ?', [busRouteId]);
-        stops.rows.forEach(row => {
-            const stopItem: StopItem = {
-                stopId: row.stop_id,
-                stopName: row.stop_name,
-            };
-            stopItems.push(stopItem);
-        });
-        return stopItems;
-    } finally {
-        await db.close();
-    }
+    const stopItems: StopItem[] = [];
+    const stops = await db.execute('SELECT rs.stop_id, stops.stop_name FROM route_stops rs JOIN stops ON rs.stop_id = stops.stop_id WHERE route_id = ?', [busRouteId]);
+    stops.rows.forEach(row => {
+        const stopItem: StopItem = {
+            stopId: row.stop_id,
+            stopName: row.stop_name,
+        };
+        stopItems.push(stopItem);
+    });
+    return stopItems;
 }
 
 function RouteScreen({navigation, route}: { navigation: any; route: any }): React.JSX.Element {

--- a/src/StopScreen.tsx
+++ b/src/StopScreen.tsx
@@ -11,61 +11,57 @@ const DEFAULT_ROUTES: {routeId: string, routeName: string, routeTimes: string[]}
 
 async function getRoutes(stopId: string) {
     const db = await openDatabase({ name: 'app.db' });
-    try {
-        const routeItems: { routeId: string, routeName: string, routeTimes: string[] }[] = [];
-        // Combined query to get routes and their arrival times for this stop
-        const sql = `
-            SELECT
-                r.route_id,
-                r.route_long_name,
-                st.arrival_time
-            FROM routes r
-            JOIN trips t ON r.route_id = t.route_id
-            JOIN stop_times st ON t.trip_id = st.trip_id
-            WHERE st.stop_id = ?
-            ORDER BY r.route_id, st.arrival_time ASC
-        `;
-        const res = await db.execute(sql, [stopId]);
+    const routeItems: { routeId: string, routeName: string, routeTimes: string[] }[] = [];
+    // Combined query to get routes and their arrival times for this stop
+    const sql = `
+        SELECT
+            r.route_id,
+            r.route_long_name,
+            st.arrival_time
+        FROM routes r
+        JOIN trips t ON r.route_id = t.route_id
+        JOIN stop_times st ON t.trip_id = st.trip_id
+        WHERE st.stop_id = ?
+        ORDER BY r.route_id, st.arrival_time ASC
+    `;
+    const res = await db.execute(sql, [stopId]);
 
-        const now = new Date();
-        const groupedByRoute: Record<string, {name: string, times: string[]}> = {};
+    const now = new Date();
+    const groupedByRoute: Record<string, {name: string, times: string[]}> = {};
 
-        res.rows.forEach(row => {
-            if (!groupedByRoute[row.route_id]) {
-                groupedByRoute[row.route_id] = { name: row.route_long_name, times: [] };
+    res.rows.forEach(row => {
+        if (!groupedByRoute[row.route_id]) {
+            groupedByRoute[row.route_id] = { name: row.route_long_name, times: [] };
+        }
+        groupedByRoute[row.route_id].times.push(row.arrival_time);
+    });
+
+    for (const routeId in groupedByRoute) {
+        const { name, times } = groupedByRoute[routeId];
+        const routeTimes: string[] = [];
+
+        for (const arrivalTime of times) {
+            const date = getArrivalDate(arrivalTime);
+            if (date > now) {
+                routeTimes.push(date.toLocaleTimeString());
             }
-            groupedByRoute[row.route_id].times.push(row.arrival_time);
-        });
-
-        for (const routeId in groupedByRoute) {
-            const { name, times } = groupedByRoute[routeId];
-            const routeTimes: string[] = [];
-
-            for (const arrivalTime of times) {
-                const date = getArrivalDate(arrivalTime);
-                if (date > now) {
-                    routeTimes.push(date.toLocaleTimeString());
-                }
-            }
-
-            if (routeTimes.length === 0 && times.length > 0) { //next trip is tomorrow
-                const earliestArrivalTime = times[0];
-                const earliestArrivalTimeTomorrow = getArrivalDate(earliestArrivalTime);
-                earliestArrivalTimeTomorrow.setDate(earliestArrivalTimeTomorrow.getDate() + 1);
-                routeTimes.push(earliestArrivalTimeTomorrow.toLocaleString());
-            }
-
-            routeItems.push({
-                routeId,
-                routeName: name,
-                routeTimes
-            });
         }
 
-        return routeItems;
-    } finally {
-        await db.close();
+        if (routeTimes.length === 0 && times.length > 0) { //next trip is tomorrow
+            const earliestArrivalTime = times[0];
+            const earliestArrivalTimeTomorrow = getArrivalDate(earliestArrivalTime);
+            earliestArrivalTimeTomorrow.setDate(earliestArrivalTimeTomorrow.getDate() + 1);
+            routeTimes.push(earliestArrivalTimeTomorrow.toLocaleString());
+        }
+
+        routeItems.push({
+            routeId,
+            routeName: name,
+            routeTimes
+        });
     }
+
+    return routeItems;
 }
 
 function StopScreen({navigation, route}: { navigation: any; route: any }): React.JSX.Element {

--- a/src/StopsScreen.tsx
+++ b/src/StopsScreen.tsx
@@ -25,8 +25,6 @@ async function getStopItems():Promise<StopItem[]>{
         }
         stopItems.push(stopItem);
     });
-    await db.close();
-
     return stopItems;
 }
 
@@ -41,7 +39,7 @@ function StopsScreen({navigation}: any): React.JSX.Element {
           setStops(data);
       }
       loadData();
-  })
+  }, []);
 
   if (!stops || stops.length === 0) {
     return (

--- a/src/gtfs/service.ts
+++ b/src/gtfs/service.ts
@@ -125,6 +125,5 @@ export async function hasAnyData(feedKey: string): Promise<boolean> {
   const db = await openDatabase({ name: 'app.db' });
   // Fast existence probe against routes (could use feed_meta too)
   const res = await db.execute('SELECT 1 AS x FROM routes WHERE feed_key = ? LIMIT 1', [feedKey]);
-  await db.close();
   return Array.isArray(res.rows) && res.rows.length > 0;
 }

